samples/rpc-priority-queryPartitions.js

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sample-metadata:
+//  title: Create and Execute Query partitions with RPC priority
+//  usage: node rpc-priority-queryPartitions.js <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
+
+'use strict';
+
+async function main(instanceId, databaseId, projectId) {
+  // [START spanner_rpc_priority_query_partitions]
+  // Imports the Google Cloud client library.
+  const {Spanner, protos} = require('@google-cloud/spanner');
+  const Priority = protos.google.spanner.v1.RequestOptions.Priority;
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  async function createAndExecuteQueryPartitionsWithRpcPriority(
+    instanceId,
+    databaseId
+  ) {
+    // Gets a reference to a Cloud Spanner instance and database
+    const instance = spanner.instance(instanceId);
+    const database = instance.database(databaseId);
+    const [transaction] = await database.createBatchTransaction();
+
+    const [partitions] = await transaction.createQueryPartitions({
+      sql: 'SELECT * FROM Singers',
+      requestOptions: {
+        priority: Priority.PRIORITY_LOW,
+      },
+    });
+    console.log(`Successfully created ${partitions.length} query partitions.`);
+
+    let row_count = 0;
+    const promises = [];
+    partitions.forEach(partition => {
+      promises.push(
+        transaction.execute(partition).then(results => {
```

Comment:
It would make more sense to set a priority for this RPC invocation, as this is something that could consume a lot of resources on the server.

Reply:
Adding priority to ```transaction.execute(partition)``` is not supported. Each partition query will inherit the options set from the partitionQuery call.

Reply:
See above. I think you're right that the `partition` will inherit the options set for the `createQueryPartitions` call, but that only applies to partitions that are used in the same client.

Reply:
Added the priority in transaction.execute(partition)